test/Microsoft.VisualStudio.Threading.Tests/JoinableTaskTests.cs

```diff
@@ -4116,6 +4116,44 @@ public void GetAwaiterCompletedTDoesNotLock()
         lockHolder.Wait();
     }
 
+    [Fact]
+    public void JoinableTaskCleanUpDependenciesForLoopDependencies()
+    {
+        this.SimulateUIThread(async delegate
+        {
+            var joinableTaskCollection = new JoinableTaskCollection(this.context, refCountAddedJobs: true);
+            JoinableTaskFactory taskFactory = this.context.CreateFactory(joinableTaskCollection);
+            ((DerivedJoinableTaskFactory)taskFactory).AssumeConcurrentUse = true;
+
+            bool isMainThreadBlockedResult = true;
```

Comment:
`isMainThreadBlockedResult` shouldn't be assigned here I think. You unconditionally assign it later, so the value here doesn't matter, and what you're testing is the later assignment. By *removing* the assignment here, you make that clear in code, and if the assignment below were ever changed by accident to not always happen, the compiler can help you realize it, but only if you remove the assignment on this line.